proofs/eo/cpc/programs/Nfa.eo

```diff
@@ -0,0 +1,161 @@
+          )
+    )
+
+    (($add_to_nexts (nfa.trans.list t tt) st nexts) (eo::ite ($in_list nexts (nfa.trans.list t st))
+                                                            ($add_to_nexts tt st nexts)
+                                                            ($add_to_nexts tt st (eo::cons nfa.trans.list (nfa.trans.list t st) nexts))))
+  )
+)
+
+; starts with set of starting states, then for every char in the string gets the set of next states
+(program $match ((c String) (s String) (nexts nfa.trans) (tt nfa.trans :list) (t nfa.trans)
+                (st nfa.trans) (z0 Int) (z1 Int))
+  (nfa.char String nfa.trans nfa.trans) nfa.trans ; current_char rest_string current_states next_states transitions -> end_states
+  (
+    (($match c s nfa.trans.null nfa.trans.null) nfa.trans.null)
+
+    (($match c "" nfa.trans.null nexts) nexts)
+
+    (($match "" s t nfa.trans.null) ($match "" s nfa.trans.null ($add_to_nexts t nfa.trans.null nfa.trans.null)))
+
+    (($match c s nfa.trans.null nexts) ($match (eo::extract s 0 0) (eo::extract s 1 (eo::len s)) nexts nfa.trans.null))
+
+    (($match c s (nfa.trans.list (nfa.trans.list (nfa.trans.init c t) st) tt) nexts)  
+            ($match c s tt ($add_to_nexts ($as_list t) st nexts))
+    )
+
+    (($match c s (nfa.trans.list (nfa.trans.list (nfa.trans.init nfa.allchar t) st) tt) nexts)  
+            ($match c s tt ($add_to_nexts ($as_list t) st nexts))
+    )
+
+    (($match c s (nfa.trans.list (nfa.trans.list (nfa.trans.init (nfa.range z0 z1) t) st) tt) nexts)
+            (eo::ite (eo::and ($compare_geq (eo::to_z c) z0) ($compare_geq z1 (eo::to_z c)))  
+            ($match c s tt ($add_to_nexts ($as_list t) st nexts))
+            ($match c s tt nexts))
+    )
+
+    (($match c s (nfa.trans.list t tt) nexts)  ($match c s tt nexts))
+  )
+)
+
+(program $is_extended ((r RegLan) (rr RegLan :list))
+  (RegLan) Bool
+  (
+    (($is_extended (re.comp r)) true)
+    (($is_extended (re.inter r rr)) true)
+    (($is_extended (re.union r rr)) (eo::or ($is_extended r) ($is_extended rr)))
+    (($is_extended (re.++ r rr)) (eo::or ($is_extended r) ($is_extended rr)))
+    (($is_extended (re.* r)) ($is_extended r))
+    (($is_extended (re.+ r)) ($is_extended r))
+    (($is_extended r) false)
+  )
+)
+
+; main function
+(program $str_eval_str_in_re_nfa ((r RegLan) (s String))
+  (String RegLan) Bool
+  (
+    (($str_eval_str_in_re_nfa s r)  
+                                    (eo::ite ($is_extended r)
+                                    ($str_eval_str_in_re s r) ; not supported - fallback to another algorithm
+                                    ($in_list ($match "" s ($get_trans r) nfa.trans.null) (nfa.trans.list nfa.trans.null nfa.trans.null)))
+                                    
+    )
+  )
+)
```

Comment:
I am curious why this PR doesn't have any changes in the cpp files -- I suggest we add an option in cvc5 that would switch between this signature and the original one.